src/CircularList/CircularlArrayFP.ts

```diff
@@ -0,0 +1,69 @@
+/**
+**
+Functional programming implementation of Circular array
+**
+**/
+
+import ICircular from "./ICircular";
+
+let strArrData: readonly string [] = [];
+let numCurrent: number = 0;
+let numCounter: number = 0;
+
+const updateWithItems = (newDataArr: string []) => {
+    strArrData = [...newDataArr];
+    numCounter = newDataArr.length;
+}
+
+const insertAtFront = (newUrl: string): string => {
+    numCounter++;
+    strArrData = [newUrl, ...strArrData];
```

Comment:
1. We should use deconstruction wisely as the performance is often bad
2. strArrData seems a bit odd, if it has to be a global stateful variable, it doesn't need to be reassigned everytime. If it doesnt, and you wanna keep FP's purity, we shall make it a parameter. (Also good for unit testing)
```
const insertAtFront = (strArrData: string [], newUrl: string)
```